### PR TITLE
Fix Merging of variables 

### PR DESCRIPTION
**Pull request by sliu008**
### Description

- Bug when we try to merge granules in collection that have day and night granules. Some granules would have extra variables in it, i believe night had more than day.
- When merging not all variables might be included really depends on the list of granules and which is first.
- Add harmony deployment step to github actions

Example: 
- if day granule was first on the list then not all the variables were included
- if night granule was first on the list then all variables were included

### Overview of work done

- Fixed it so that if theres a difference in the variables between granules then update the var info list

### Overview of verification done

- tested local concatenation of MODIS_T-JPL-L2P-v2019.0

### Overview of integration done

_Explain how this change was integration tested. Provide screenshots or logs if appropriate. An example of this would be a local Harmony deployment._

## PR checklist:

* [ ] Linted
* [ ] Updated unit tests
* [ ] Updated changelog
* [ ] Integration testing

_See [Pull Request Review Checklist](../CONTRIBUTING.md#reviewing) for pointers on reviewing this pull request_

### CODE REVIEW

File: deployment/harmony_deploy.py
Comment (frankinspace) [inline deployment/harmony_deploy.py:1]: We should consider turning this into a github action

Comment (joshgarde) [reply deployment/harmony_deploy.py:1]: I'm unclear why it has to be a Python script specifically. Doesn't seem like anything that can't be done within bash + curl

**Review by joshgarde — APPROVED** [review]
Fix looks good; always beautiful to see a bugfix where code can be removed/rewritten to be more elegant

File: deployment/harmony_deploy.py
Comment (sliu008) [reply deployment/harmony_deploy.py:1]: yea it could be i just wrote it in python

**Pull request merged**